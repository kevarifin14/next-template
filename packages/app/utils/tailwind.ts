export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
