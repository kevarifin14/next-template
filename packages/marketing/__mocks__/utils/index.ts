import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
