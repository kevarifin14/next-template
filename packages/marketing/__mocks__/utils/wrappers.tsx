import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
