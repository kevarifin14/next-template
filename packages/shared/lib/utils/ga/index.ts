export const pageview = (url) => {
  window.gtag("config", process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
    page_path: url,
  });
};

type EventCategory =
  | "lead"
  | "outbound link"
  | "call to action"
  | "login"
  | "swap";

export const event = (
  category: EventCategory,
  action,
  label = "",
  params = {}
) => {
  if (process.env.NODE_ENV === "production") {
    window.gtag("event", action, {
      event_category: category,
      event_label: label,
      ...params,
    });
  }
};
