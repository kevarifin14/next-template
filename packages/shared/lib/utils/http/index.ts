import QueryString from 'query-string';

export function generateQueryString(params) {
  return QueryString.stringify(params);
}

export function generateEndpoint(endpoint, params) {
  if (Object.keys(params).length === 0) {
    return endpoint;
  }
  return `${endpoint}?${generateQueryString(params)}`;
}

export const get = async (endpoint, params = {}) => {
  const endpointWithParams = generateEndpoint(endpoint, params);
  const response = await fetch(endpointWithParams);

  if (response.status === 200) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};

export const post = async (endpoint, body, headers?) => {
  const response = await fetch(
    endpoint,
    {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
        ...headers,
      },
    },
  );

  if (response.status >= 200 && response.status < 400) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};
