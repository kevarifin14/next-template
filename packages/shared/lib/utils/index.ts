export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
