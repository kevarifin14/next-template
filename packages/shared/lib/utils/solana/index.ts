/* istanbul ignore file */
import {
  AccountLayout,
  u64,
  AccountInfo as TokenAccountInfo,
  MintInfo,
} from "@solana/spl-token";
import { PublicKey } from "@solana/web3.js";

export const deserializeTokenAccount = (address: PublicKey, data: Buffer) => {
  const {
    mint,
    owner,
    amount,
    delegateOption,
    delegate,
    delegatedAmount,
    state,
    isNative,
    isNativeOption,
    closeAuthority,
    closeAuthorityOption,
  } = AccountLayout.decode(data);

  const accountInfo: TokenAccountInfo = {
    address,
    mint: new PublicKey(mint),
    owner: new PublicKey(owner),
    amount: u64.fromBuffer(amount),
    isInitialized: state !== 0,
    isFrozen: state === 2,

    ...(delegateOption === 0
      ? { delegate: null, delegatedAmount: new u64(0) }
      : {
          delegate: new PublicKey(delegate),
          delegatedAmount: u64.fromBuffer(delegatedAmount),
        }),

    ...(isNativeOption === 1
      ? { rentExemptReserve: u64.fromBuffer(isNative), isNative: true }
      : { rentExemptReserve: null, isNative: false }),

    ...(closeAuthorityOption === 0
      ? { closeAuthority: null }
      : { closeAuthority: new PublicKey(closeAuthority) }),
  };

  return accountInfo;
};

export const diffTokenAccountInfoAmounts = (
  prevTokenAccountInfo: TokenAccountInfo,
  tokenAccountInfo: TokenAccountInfo
) => {
  if (!prevTokenAccountInfo) {
    return tokenAccountInfo.amount;
  }

  return tokenAccountInfo.amount.sub(prevTokenAccountInfo.amount);
};

export const fromLamports = (
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo
) => {
  if (!tokenAccountInfo) {
    return 0;
  }

  const amount =
    typeof tokenAccountInfo === "number"
      ? tokenAccountInfo
      : tokenAccountInfo.amount.toNumber();

  const precision = 10 ** (mintInfo?.decimals || 9);
  return amount / precision;
};

export function formatTokenAmount(
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo,
  prefix = "",
  suffix = "",
  precision = 3
): string {
  return `${[prefix]}${fromLamports(tokenAccountInfo, mintInfo).toFixed(
    precision
  )}${suffix}`;
}

export const formatDollar = (value) => {
  const formatter = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
  });

  return formatter.format(value);
};
