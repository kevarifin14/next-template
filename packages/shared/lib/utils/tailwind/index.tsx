import { ComponentProps } from 'react';

export type TailwindSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl';

export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};

export const sizeToClassName = (size: TailwindSize) => {
  const dimension = sizeToDimension[size];
  return `h-${dimension} w-${dimension}`;
};

export type AlertType = 'success' | 'info' | 'error' | 'warning';

export type Heroicon = (props: ComponentProps<'svg'>) => JSX.Element; // eslint-disable-line