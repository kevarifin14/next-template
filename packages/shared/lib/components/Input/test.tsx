import { render } from '@testing-library/react';
import React from 'react';

import { Input } from '.';
import { errorTypeToMessage } from '../../utils/forms';

describe('Input', () => {
  const mockLabel = 'mockLabel';

  it('matches snapshot', () => {
    const { baseElement } = render(<Input />);
    expect(baseElement).toMatchSnapshot();
  });

  describe('Input Size', () => {
    it('renders lg input', () => {
      const { getByRole } = render(<Input size="lg" />);
      expect(getByRole('textbox')).toHaveClass('px-5 py-3 text-lg');
    });

    it('renders xl input', () => {
      const { getByRole } = render(<Input size="xl" />);
      expect(getByRole('textbox')).toHaveClass('px-6 py-4 text-xl');
    });
  });

  it('shows message on error', () => {
    const { queryByText } = render(<Input error={{ type: 'required' }} />);

    expect(queryByText(errorTypeToMessage('required'))).toBeTruthy();
  });

  it('shows default error message', () => {
    const { queryByText } = render(<Input error={{ type: 'no type' }} />);

    expect(queryByText(errorTypeToMessage('no type'))).toBeTruthy();
  });

  it('shows custom error message', () => {
    const mockCustomError = 'Wrong';
    const { queryByText } = render(
      <Input label={mockLabel} error={{ message: mockCustomError }} />,
    );

    expect(queryByText(mockCustomError)).toBeTruthy();
  });

  it('renders label', () => {
    const { getByLabelText } = render(<Input label={mockLabel} />);
    expect(getByLabelText).toBeTruthy();
  });
});
