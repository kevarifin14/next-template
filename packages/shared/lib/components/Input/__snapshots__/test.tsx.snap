// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full"
    >
      <div
        class="relative"
      >
        <input
          class="shadow-sm focus:outline-none w-full border bg-light border-light-400 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary-600 focus:ring-primary-600 focus:ring-1 dark:bg-dark-light dark:border-transparent dark:text-white px-4 py-2"
          id="123456879"
        />
      </div>
    </div>
  </div>
</body>
`;
